### PR TITLE
HttpServer handles Json response

### PR DESCRIPTION
**Pull request by fi-ya**
This PR relates to [this ticket](https://trello.com/c/GSZQMYrI/98-scenario-returning-a-json-response)

**Given** I make a GET request to "/json_response"
**Then** my response should have status code 200
**And** my response should return JSON
**And** my response should have a JSON body

### CODE REVIEW

**Review by devlinmorrow — APPROVED** [review]
LGTM, some comments

File: src/test/java/org/httpserver/handler/JsonHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    void returnsResponse_withStatusLineAndBody_withJson() throws JsonProcessingException {
+        RequestLine mockRequestLine = new RequestLine(GET, "/json_response", "HTTP/1.1");
+        JsonHandler jsonHandler = new JsonHandler();
+
+        Response actualResponse = jsonHandler.handleResponse(new Request(mockRequestLine, new LinkedHashMap<>(), ""));
+
+        assertEquals("HTTP/1.1 200 OK\r\n", actualResponse.getResponseStatusLine());
+        assertTrue(actualResponse.getResponseHeaders().contains("Content-Type"));
+        assertTrue(actualResponse.getResponseHeaders().contains("application/json;charset=utf-8"));
+        assertEquals("{\"key1\":\"value1\",\"key2\":\"value2\"}", new String(actualResponse.getBodyBytes()));
```
[the remark this answers is not in the document]
Comment (fi-ya) [reply src/test/java/org/httpserver/handler/JsonHandlerTest.java:35]:
```  
assertEquals(new HashMap() {{
            put("key1", "value1");
            put("key2", "value2");
        }}, new ObjectMapper().readValue(actualResponse.getBodyBytes(), new TypeReference<>(){}));
```

i deserialized `actualResponse.getBodyBytes()` using `ObjectMapper` Class method `readValue()`

[the remark this answers is not in the document]
Comment (fi-ya) [reply src/test/java/org/httpserver/handler/JsonHandlerTest.java:35]: Is this what you meant?

**Review by andy8thlight — APPROVED** [review]
As Devlin has already mentioned, the "printlns" need removing and the comment about deserialisation in the test I agree with. Apart from this (and the obvious hard-coding of the JSON response), this looks ok.

**Pull request merged**